### PR TITLE
Bugs 6947 and 6974 - Handle Socket Fails

### PR DESCRIPTION
**Pull request by matthopson**
This attempts to instantiate the socket connection BEFORE any interactors / persistence methods are called for socket-dependent operations. A modal is thrown if the socket fails to open, allowing the user to try again or seek support, without side-effects in the application.

Screen Shot 2020-11-06 at 2 13 02 PM

ustaxcourt/ef-cms#7460 ustaxcourt/ef-cms#7449

### CODE REVIEW

**Comment by matthopson** [comment]
FYI @klohman - lemme know if we want to do any tweaks to the messaging here.

File: web-client/src/views/WebSocketErrorModal.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <ConfirmModal
+      cancelLabel="Cancel"
+      confirmLabel="Ok"
+      hasErrorState={true}
+      headerIcon="info-circle"
+      headerIconClassName="text-secondary-dark"
+      noCancel={true}
+      preventCancelOnBlur={true}
+      title="An Unexpected Error Occured"
```
Comment (rachaelparris) [inline web-client/src/views/WebSocketErrorModal.jsx]: `Occurred` I believe?

**Review by rachaelparris — APPROVED** [review]
One misspelling. Otherwise looks great! Thanks for taking care of this. I think it's a much more solid solution.

**Comment by klohman** [comment]
> FYI @klohman - lemme know if we want to do any tweaks to the messaging here.

Can we update the second line to "Try again, or contact dawson.support@ustaxcourt.gov for assistance." 
Keeps it more consistent with our other error messages

**Pull request merged**